src/main/scala/net/ripe/rpki/nro/service/Ports.scala

```diff
@@ -212,9 +230,30 @@ object Ports extends Logging {
     }
   }
 
-  def readConflicts(conflictFile: String = s"${config.previousConflictFile}"): List[Conflict] = {
-    logger.debug(s"Reading conflicts from ${config.previousConflictFile}")
-    Using.resource(CSVReader.open(new File(conflictFile))){ reader =>
+  private def fetchAndWriteLocally(source: String, dest: String): Unit = {
+    val fetchResponse: Future[Response] = fetchWithRetries(source)
+    Try(Await.result(fetchResponse, Duration.Inf)) match {
+      case Success(response) if response.statusCode == 200 =>
+        val responseText = response.text()
+        Using.resource(new PrintWriter(new File(dest))) { writer =>
+          writer.write(responseText)
+          logger.info(s"---Done fetching $source into $dest---\n\n\n")
+        }
+      case _ =>
+        logger.error(s"Failed to fetch $source after $maxRetries retries")
+        System.exit(1)
```

Comment:
Having `System.exit` in the middle of application logic is often confusing. Consider using return values (`Try`, `Either`, `Future`, and the likes) to indicate errors and handle them in main.